### PR TITLE
Add Fantomas tool and format F# code

### LINKED ISSUE #434: Fantomas code formatter
Add the [Fantomas code formatter](https://github.com/fsprojects/fantomas/#fantomas), which is now recommended by the [F# code formatting guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#use-an-automatic-code-formatter).

### PR DESCRIPTION
**Pull request by LyndonGingerich**
Resolves #434

Should I examine any particular settings to minimize changes?

### CODE REVIEW

**Comment by LyndonGingerich** [comment]
Do we want a build tool to ensure that merged code is formatted correctly?

**Comment by TysonMN** [comment]
> Do we want a build tool to ensure that merged code is formatted correctly?

I think that would be good.

Are the formatting choices of Fantomas configurable?

**Comment by LyndonGingerich** [comment]
> Are the formatting choices of Fantomas configurable?

Very. https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html

**Comment by rfreytag** [comment]
Had not seen Fantomas before.  So, I looked at Fantomas looking for some standardization of blank line practices like skipping extra, extra lines after Modules and _could not find that._  Did I miss it?  Since you folks like that practice you might want to customize your usage of Fantomas to include that practice.

**Comment by LyndonGingerich** [comment]
@rfreytag I don't think that Fantomas removes newlines except at the ends of files.

File: src/Elmish.WPF.Tests/BindingTests.fs [hunk 1]
```diff
@@ -80,11 +85,12 @@ module oneWay =
...
   [<Fact>]
   let ``final get returns value from original get`` () =
-    Property.check <| property {
+    Property.check
+    <| property {
       let! x = GenX.auto<int>
 
       let get = string<int>
-      let d = Binding.oneWay(get) |> getOneWayData
+      let d = Binding.oneWay (get) |> getOneWayData
```
Comment (TysonMN) [inline src/Elmish.WPF.Tests/BindingTests.fs:93]: Can there be no space between a method and the opening parenthesis?

**Review by TysonMN — CHANGES_REQUESTED** [review]

**Comment by LyndonGingerich** [comment]
@TysonMN Should I look into a GitHub action? It would take a while, and I don't (and probably shouldn't) have the access required to create one here.

> Are the formatting choices of Fantomas configurable?

Did you want any specific configuration?

**Comment by LyndonGingerich** [comment]
@rfreytag There is a [setting](https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html#fsharp_keep_max_number_of_blank_lines) for the maximum number of consecutive blank lines, but it doesn't distinguish between blank lines after modules and blank lines elsewhere.

File: src/Elmish.WPF.Tests/BindingTests.fs (hunk 1 above)
Comment (TysonMN) [reply src/Elmish.WPF.Tests/BindingTests.fs:93]: That makes sense. Thanks for the information.

File: src/Elmish.WPF.Tests/BindingVmHelpersTests.fs [hunk 2]
```diff
@@ -106,15 +115,19 @@ module Set =
...
       let! initialModel = g
       let! newModel = g |> GenX.notEqualTo initialModel
 
       let model = ref initialModel
-      let dispatch msg = failwith $"Should not dispatch message {msg}"
-      let binding =
-        BindingData.TwoWay.id
+
+      let dispatch msg =
+        failwith $"Should not dispatch message {msg}"
```
Comment (TysonMN) [inline src/Elmish.WPF.Tests/BindingVmHelpersTests.fs:127]: What is the setting for maximum line length?  This line seems plenty short and could stay on one line.

**Review by TysonMN — CHANGES_REQUESTED** [review]

File: src/Elmish.WPF.Tests/BindingVmHelpersTests.fs (hunk 2 above)
Comment (LyndonGingerich) [reply src/Elmish.WPF.Tests/BindingVmHelpersTests.fs]: Fixed.

**Comment by LyndonGingerich** [comment]
I've let this branch grow old. I'll make another rather than dealing with merge conflicts.

**Pull request closed without merging**